src/ui/simulator/application/study.cpp

```diff
@@ -286,7 +286,7 @@ class JobSaveStudy final : public Toolbox::Jobs::Job
 
 public:
     //! Reference to the study to save
-    Data::Study& study;
+    Data::Study::Ptr study;
```

Comment:
Is it better to replace a & to the study with a smart pointer to the study ?

Reply:
It's better yes : keeping a `shared_ptr` prevents accidental deletion. Keeping only a reference to the managed object (of type `Study`) could lead to situations equivalent to

```cpp
Study* study = new Study();
Study* ref = study;
delete study;
ref->doSomething(); // Oops !
```

The only difference is that `shared_ptr` may hide this behavior.